providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -814,6 +814,16 @@ static void *dh_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
     DH_clear_flags(dh, DH_FLAG_TYPE_MASK);
     DH_set_flags(dh, gctx->dh_type);
 
+#ifdef FIPS_MODULE
+    if (!ossl_fips_self_testing()) {
+        ret = ossl_dh_check_pairwise(dh, 1);
```

Comment:
I'm wondering if the `1` as the second param here is really correct. It causes this test to not fail if any of `p`, `g`, `priv_key` and `pub_key` are NULL. But is there a valid scenario where they can be NULL here?